eth/api.go

```diff
@@ -678,3 +678,70 @@ func (api *PrivateDebugAPI) GetAccessibleState(from, to rpc.BlockNumber) (uint64
 	}
 	return 0, fmt.Errorf("No state found")
 }
+
+type PublicWemixAPI struct {
+	e *Ethereum
+}
+
+func NewPublicWemixAPI(e *Ethereum) *PublicWemixAPI {
+	return &PublicWemixAPI{e}
+}
+
+func (api *PublicWemixAPI) BriocheConfig() *params.BriocheConfig {
+	return api.e.BlockChain().Config().Brioche
+}
+
+type HalvingInfo struct {
+	HalvingTimes uint64   `json:"halvingTimes"`
+	StartBlock   *big.Int `json:"startBlock"`
+	EndBlock     *big.Int `json:"endBlock"`
+	BlockReward  *big.Int `json:"blockReward"`
+}
+
+func (api *PublicWemixAPI) HalvingSchedule() []*HalvingInfo {
+	bc := api.BriocheConfig()
+	if bc.FirstHalvingBlock == nil || bc.HalvingPeriod == nil || bc.HalvingTimes == 0 {
+		return nil
+	}
+
+	var lastRewardBlock *big.Int
+	if bc.FinishRewardBlock != nil {
+		lastRewardBlock = new(big.Int).Sub(bc.FinishRewardBlock, common.Big1)
+	}
+
+	result := make([]*HalvingInfo, 0)
+	for i := uint64(0); i < bc.HalvingTimes; i++ {
+		startBlock := new(big.Int).Add(bc.FirstHalvingBlock, new(big.Int).Mul(bc.HalvingPeriod, new(big.Int).SetUint64(i)))
+		if lastRewardBlock != nil && startBlock.Cmp(lastRewardBlock) == 1 {
+			break
+		}
+		result = append(result, &HalvingInfo{
+			HalvingTimes: i + 1,
+			StartBlock:   startBlock,
+			EndBlock:     new(big.Int).Sub(new(big.Int).Add(startBlock, bc.HalvingPeriod), common.Big1),
+			BlockReward:  api.GetBriocheBlockReward(startBlock),
+		})
+	}
+
+	result[len(result)-1].EndBlock = lastRewardBlock
+
+	return result
+}
+
+func (api *PublicWemixAPI) GetBriocheBlockReward(height *big.Int) *big.Int {
+	if wemixapi.Info == nil {
+		return nil
+	}
+	config := api.e.BlockChain().Config()
+	wemixInfo := *(wemixapi.Info().(*map[string]interface{}))
+
+	if height == nil {
+		height = api.e.BlockChain().CurrentHeader().Number
+	}
+
+	if config.IsBrioche(height) {
+		return config.Brioche.GetBriocheBlockReward(wemixInfo["defaultBriocheBlockReward"].(*big.Int), height)
+	} else {
+		return wemixInfo["blockReward"].(*big.Int)
+	}
```

Comment:
I think it's always a good to include code to verify that type assertions are successful. So, it seems better to add verifying code when doing type assertion of wemixInfo. Of course, it would be ideal to include code to verify for the type assertion of (*big.Int) on the subsequent line, but it doesn't seem necessary to go that far.

And it seems there might be no need to receive wemixInfo as a pointer since it doesn't need modifying wemixInfo's values. And in my understanding, since the wemixInfo data should not be modified by the function(GetBriocheBlockReward) call, using a pointer seems risky.

Therefore, if the code below works correctly, how about modifying it as follows?
```
        wemixInfo, ok := wemixapi.Info().(map[string]interface{})
	if !ok {
		return nil
	}

	if height == nil {
		height = api.e.BlockChain().CurrentHeader().Number
	}

	if config.IsBrioche(height) {
		return config.Brioche.GetBriocheBlockReward(wemixInfo["defaultBriocheBlockReward"].(*big.Int), height)
	} else {
		return wemixInfo["blockReward"].(*big.Int)
	}
```